subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -272,7 +272,7 @@ public String determineName(Configuration c) {
      * @return The name of the associated upload task
      * @see org.gradle.api.tasks.Upload
      */
-    @Deprecated
+    @Deprecated // TODO:Finalize Upload Removal - Issue #21439
```

Comment:
IIUC, if we remove this, we break compatibility with Spring Boot 2.5.x and 2.6.x but not 2.7.x. Is that right?

Reply:
No, that's wrong.  If we remove this, the latest 2.7.3 also fails.